src/components/TxFlow/ModalProvider.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  createContext,
+  type Dispatch,
+  type ReactElement,
+  type ReactNode,
+  type SetStateAction,
+  useState,
+  type ComponentProps,
+  useEffect,
+} from 'react'
+import ModalDialog from '@/components/common/ModalDialog'
+import TokenTransferFlow from '@/components/TxFlow/TokenTransfer/TokenTransferFlow'
+import RejectTx from '@/components/TxFlow/RejectTx'
+import NewTxMenu from '@/components/TxFlow/NewTx'
+import ReplaceTxMenu from '@/components/TxFlow/ReplaceTx'
+import { useRouter } from 'next/router'
+
+export enum ModalType {
+  SendTokens = 'sendTokens',
+  RejectTx = 'rejectTx',
+  ReplaceTx = 'replaceTx',
+  NewTx = 'newTx',
+}
+
+const ModalTypes = {
+  [ModalType.SendTokens]: TokenTransferFlow,
+  [ModalType.RejectTx]: RejectTx,
+  [ModalType.ReplaceTx]: ReplaceTxMenu,
+  [ModalType.NewTx]: NewTxMenu,
+}
```

Comment:
I've just realized it's probably easier to just let modal-opening buttons pass `TokenTransferFlow` directly instead of a string. That would make the dependency explicit.

Reply:
The implementation we opted for aimed to narrow the possible components we want to show in the modal. It comes at the cost of maintaining the map but setting a component in the dispatcher would make harder to reinforce the types.
We are also not sold on storing the component as state.

Reply:
You can still limit the type of components setVisibleModal accepts.